### PR TITLE
Add API functionality

### PR DESCRIPTION
**Pull request by MrBrN197**
## Changes | ⚔ 

- ✔ Refresh the scoreboard when a user clicks the refresh button
- ✔ Submit a new user and score to the API when the user clicks Submit.
- ✔ Added functions for GET and POST API functions in "./api.js".
- ✔ Resolve issue #2

### CODE REVIEW

File: src/dom-functions.js [hunk 1]
```diff
@@ -8,29 +9,48 @@ export const initializePageHTML = () => {
...
         <input type="submit" value="Submit">
       </form>
     </section>
   </main>
   `;
   document.body.innerHTML = initialHTML;
+};
+
+export const initializeScoreboard = () => {
+  const scoreboard = document.getElementById('scoreboard');
```
Comment (EricMbouwe) [inline src/dom-functions.js]: - [OPTIONAL] Prefer the use of camelCase for variable names.

File: src/listeners.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const formElem = document.querySelector('form');
+  const userInput = document.getElementById('name');
+  const scoreInput = document.getElementById('score');
+  formElem.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const submitInput = formElem.querySelector('input[type=submit]');
+    setDisabledState(submitInput, true);
+    const success = await addToScoreBoard(userInput.value, scoreInput.value);
+    if (success) initializeScoreboard();
+    setDisabledState(submitInput, false);
```
Comment (EricMbouwe) [inline src/listeners.js]: - [OPTIONAL] It can be good to reset the form fields after submitting the data with success.

File: src/api.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export const BASE_URL = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api';
+
+export const GAME_ID = 'sa7iLyWCcC3Ao7tJMfAb';
+
+export const getScores = () => fetch(`${BASE_URL}/games/${GAME_ID}/scores`)
+  .then((resp) => resp.json());
```
Comment (EricMbouwe) [inline src/api.js]: - [x] Good job done here! However, for this project, you are required to use `async` and `await` with the `fetch` method to send and receive data. Let's fix that

File: src/api.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      user: username,
+      score,
+    }),
+  },
+).then((resp) => resp.json());
```
Comment (EricMbouwe) [inline src/api.js]: - [x] Good job done here! However, for this project, you are required to use `async` and `await` with the `fetch` method to send and receive data. Let's fix that

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @MrBrN197,

Good job so far! 
- Nice code organization 💪

There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/api.js (hunk 4 above)
Comment (MrBrN197) [reply src/api.js]: Thank you I'll make these improvements.

File: src/api.js (hunk 3 above)
Comment (MrBrN197) [reply src/api.js]: Thank you I'll make these improvements.

File: src/dom-functions.js (hunk 1 above)
Comment (MrBrN197) [reply src/dom-functions.js]: Hi, Thank you although scoreboard is one word

File: src/listeners.js (hunk 2 above)
Comment (MrBrN197) [reply src/listeners.js]: Thank. You. I'll work on this

**Review by dasileker — APPROVED** [review]
### **Hi @MrBrN197,** 

### **Congratulations! 🎉 Project Approved ✅**


Approved


Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**